daemon/scd.c

```diff
@@ -74,21 +74,26 @@ scd_fork_and_exec(void)
 }
 
 int
-scd_init(void)
+scd_init(bool hostedmode_active)
 {
 	int sock = -1;
 
-	// if device.cert is not present, start scd to initialize device (provisioning mode)
-	if (!file_exists(DEVICE_CERT_FILE)) {
-		INFO("Starting scd in Provisioning / Installing Mode");
-		// Start the SCD in provisioning mode
-		const char *const args[] = { SCD_BINARY_NAME, NULL };
-		IF_FALSE_RETVAL_TRACE(proc_fork_and_execvp(args) == 0, -1);
-	}
+	// In hosted mode, the init scripts should take care of correct scd initialization and start-up
+	if (!hostedmode_active) {
+		// if device.cert is not present, start scd to initialize device (provisioning mode)
+		if (!file_exists(DEVICE_CERT_FILE)) {
+			INFO("Starting scd in Provisioning / Installing Mode");
+			// Start the SCD in provisioning mode
+			const char *const args[] = { SCD_BINARY_NAME, NULL };
+			IF_FALSE_RETVAL_TRACE(proc_fork_and_execvp(args) == 0, -1);
+		}
 
-	// Start SCD and wait for control interface
-	scd_pid = scd_fork_and_exec();
-	IF_TRUE_RETVAL_TRACE(scd_pid == -1, -1);
+		// Start SCD and wait for control interface
+		scd_pid = scd_fork_and_exec();
```

Comment:
scd_pid is not set in case of hostedmode_active == true. Thus default value -1 is used in cleanup.
Which results in a kill(-1, SIGTERM), killing all processes of the host, if cmld is terminating.
Thus, check if scd_pid == -1 and return early in scd_cleaup() similar to tss_cleanup().